From 548ebf35f252116c6020d745788bf7b74a6cbe8e Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Thu, 13 Feb 2020 18:03:51 +0200
Subject: [PATCH] [SW-0] goya2: update public key verification condition

There's no need to check which public key were last
verified in case no valid public keys were found.

Change-Id: Ia3f8acda6e58fdd7c2026ff0a3f30e4253aa200f
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/common/secureboot.c b/common/secureboot.c
index 6a275f6..0f3c3ae 100644
--- a/common/secureboot.c
+++ b/common/secureboot.c
@@ -74,7 +74,7 @@
 			break;
 	}
 	/* if for loop doesn't find a valid key hash in fuse, return error */
-	if (!verify && (key_scan == SECUREBOOT_HABANA_KEY)) {
+	if (!verify) {
 		regmap_writel(err, mmPSOC_SECURITY_SW_STATUS_0);
 		regmap_writel(verify, mmPSOC_SECURITY_SW_STATUS_1);
 		return SECUREBOOT_VERIFY_FAILED;
